### PR TITLE
[WFLY-7186] Add a new quickstart demonstrating EE security with WildFly Elytron.

### PR DESCRIPTION
**Pull request by darranl**

### CODE REVIEW

**Comment by sgilda** [comment]
@darranl : I made a few fixes and edits and issued a PR back to your branch. Let me know if you have any questions.

File: ee-security/README.adoc [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+< X-MESSAGE: Please resubmit the request with a username specified using the X-USERNAME and a password specified using the X-PASSWORD header.
+----
+
+This first request shows the client is being promted to authenticate, the `X-MESSAGE` header is providing additional information as to how the client can achieve this.
+
+The request can now be submitted with the previously added user.
+
+[source,options="nowrap"]
+----
+$ curl -v http://localhost:8080/ee-security/secured -H "X-Username:darranl" -H "X-Password:password1!"
```
Comment (sgilda) [inline ee-security/README.adoc]: This line should be: 

$ curl -v http://localhost:8080/ee-security/secured -H "X-Username:quickstartUser" -H "X-Password:quickstartPwd1!"

File: ee-security/README.adoc [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+Identity as available from SecurityContext 'quickstartUser'
+Identity as available from injection 'quickstartUser'
+----
+
+The resulting output shows authentication was successful and the correct identity has been established.
+
+// Undeploy the Quickstart
+include::../shared-doc/undeploy-the-quickstart.adoc[leveloffset=+1]
+
+The following CLI command will then revert the configuration changes made previously.
```
Comment (sgilda) [inline ee-security/README.adoc]: You could define this attribute near the top of the file:

{code}
:restoreScriptName: restore-configuration.cli
{code}
And then replace these lines with:
{code}
// Restore the {productName} Standalone Server Configuration
include::../shared-doc/restore-standalone-server-configuration.adoc[leveloffset=+1]

// Restore the {productName} Standalone Server Configuration Manually
include::../shared-doc/restore-standalone-server-configuration-manual.adoc[leveloffset=+2]
{code}

File: ee-security/README.adoc (hunk 1 above)
Comment (sgilda) [inline ee-security/README.adoc]: This line should be: 
```
$ curl -v http://localhost:8080/ee-security/secured -H "X-Username:quickstartUser" -H "X-Password:quickstartPwd1!"
```

File: ee-security/README.adoc (hunk 2 above)
Comment (sgilda) [inline ee-security/README.adoc]: You could define this attribute near the top of the file:
```
:restoreScriptName: restore-configuration.cli
```

And then replace this line and the lines that follow with:
```
// Restore the {productName} Standalone Server Configuration
include::../shared-doc/restore-standalone-server-configuration.adoc[leveloffset=+1]

// Restore the {productName} Standalone Server Configuration Manually
include::../shared-doc/restore-standalone-server-configuration-manual.adoc[leveloffset=+2]
```

File: ee-security/README.adoc (hunk 1 above)
Comment (emmartins) [reply ee-security/README.adoc]: Will that work on Windows?

Comment (darranl) [reply ee-security/README.adoc]: They should be able to install curl on Windows.

Comment (sgilda) [reply ee-security/README.adoc]: Yes it should. These quickstarts, which have been around a very long time, also use the `curl` command:

- tasks-rs
- helloworld-html5
- http-custom-mechanism

**Comment by emmartins** [comment]
Ok, didn’t noticed that.

File: ee-security/README.adoc (hunk 1 above)
Comment (ctomc) [reply ee-security/README.adoc]: On windows in powershell, the curl command could be rewritten as:
`curl -v http://localhost:8080/ee-security/secured -H  @{"X-Username"="quickstartUser";"X-Password"="quickstartPwd1!"}`

curl is alias to Invoke-WebRequest which has lots of options, but it is not 1:1 mapping to curl

**Comment by sgilda** [comment]
@emmartins : I have created a replacement PR if you want to close this one: https://github.com/wildfly/quickstart/pull/329

**Pull request merged**